### PR TITLE
Add template rancher backup for cattle

### PR DESCRIPTION
**Pull request by disaster37**
It's a general purpose to solve backup matter on Rancher

This template deploy one container that manage backup aspect in your Rancher environment.
For the moment, this template permit to perform the following actions:

- Dump automatically all databases supported before start the remote backup:
  - PostgreSQL
  - MySQL
  - MariaDB
  - MongoDB
  - Elasticsearch
- Dump automatically all rancher stacks settings (docker-compose.yml and rancher-compose.yml) before start the remote backup
- Perform the remote backup of all /backup with duplicity (you can mount your shares storages on /backup to backup them)

We use the power of DIND (Docker in docker) to get the needed tools to perform dump, and then we use the power of duplicity to perform a remote Backup.
All auto discover work with the power of Rancher API

PS : I upgrade rancher-backup from the previous PR to support new rancher version, so you can close #374

### CODE REVIEW

File: templates/rancher-backup/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    CLOUDFILES_APIKEY: ${CLOUDFILES_APIKEY}
+    DPBX_ACCESS_TOKEN: ${DPBX_ACCESS_TOKEN}
+    GS_ACCESS_KEY_ID: ${GS_ACCESS_KEY_ID}
+    GS_SECRET_ACCESS_KEY: ${GS_SECRET_ACCESS_KEY}
+    GOOGLE_DRIVE_ACCOUNT_KEY: ${GOOGLE_DRIVE_ACCOUNT_KEY}
+    SWIFT_USERNAME: ${SWIFT_USERNAME}
+    SWIFT_PASSWORD: ${SWIFT_PASSWORD}
+    SWIFT_AUTHURL: ${SWIFT_AUTHURL}
+  labels:
+    io.rancher.sidekicks: backup-data
```
Comment (rawmind0) [inline templates/rancher-backup/0/docker-compose.yml]: Add labels to autogenerate RANCHER_API_URL, RANCHER_API_KEY and RANCHER_API_SECRET
++    io.rancher.container.create_agent: true
++    io.rancher.container.agent.role: environment

File: templates/rancher-backup/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+backup:
+  environment:
+    CRON_SCHEDULE: ${CRON_SCHEDULE}
+    BACKEND: ${BACKEND}
+    TARGET_PATH: ${TARGET_PATH}
+    BK_FULL_FREQ: ${BK_FULL_FREQ}
+    BK_KEEP_FULL: ${BK_KEEP_FULL}
+    BK_KEEP_FULL_CHAIN: ${BK_KEEP_FULL_CHAIN}
+    VOLUME_SIZE: ${VOLUME_SIZE}
```
Comment (rawmind0) [inline templates/rancher-backup/0/docker-compose.yml]: Autogenerated RANCHER_API_URL, RANCHER_API_KEY and RANCHER_API_SECRET with labels below. Don't need to specify them.

File: templates/rancher-backup/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    - variable: DISABLE_DUMP_RANCHER
+      description: "If disabled, it's doesn't perform the backup of stack definition"
+      label: "Disable Rancher dump"
+      type: "string"
+      required: true
+      default: "false"
+      type: "enum"
+      options:
+        - "false"
+        - "true"
```
Comment (rawmind0) [inline templates/rancher-backup/0/rancher-compose.yml]: Autogenerated RANCHER_API_URL, RANCHER_API_KEY and RANCHER_API_SECRET with labels above. Don't need to ask for them.

**Comment by rawmind0** [comment]
Hi @disaster37 ...Thanks so much for your cool PR and contribute.

I've requested some changes to autogenerate RANCHER_API_URL, RANCHER_API_KEY and RANCHER_API_SECRET env variables, and don't need to ask for them.

Best regards...

**Comment by disaster37** [comment]
hi,

I have embedded your change request

Best Regards,
SL

**Comment by rawmind0** [comment]
Thanks @disaster37 ....

LGTM..

**Pull request merged**